pkgs/development/yottadb/default.nix

```
@@ -0,0 +1,225 @@
+{
+  lib,
+  config,
+  makeWrapper,
+  stdenv,
+  writeText,
+  writeScript,
+
+  cmake,
+  coreutils-full,
+  elfutils,
+  fetchFromGitLab,
+  git,
+  glibcLocales,
+  libxcrypt,
+  icu,
+  ncurses,
+  openssl,
+  pkg-config,
+  tcsh,
+  util-linux,
+  zlib
+}:
+
+# To test this package, go to $NIXPKGS/nixos/tests and run:
+#   $ $(nix-build -A driverInteractive yottadb.nix)/bin/nixos-test-driver -I
+#   >>> test_script()
+
+stdenv.mkDerivation rec {
+  pname = "yottadb";
+  version = "1.38";
+
+  src = fetchFromGitLab {
+    owner = "YottaDB";
+    repo = "DB/YDB";
+    rev = "r${version}";
+    leaveDotGit = true;
+    hash = "sha256-k6kAIdPW75cqdT+dXqjlS8TpT2NjQYBTceDzMvWD/oE=";
+  };
+
+  patches = [
+    ./01-gtmsecshr-relocate.patch
+  ];
+
+  NIX_CFLAGS_COMPILE = "-DYDB_EXTERNAL_SECSHR_PARENT_DIR=\"${ydbSecRunDir}\"";
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = buildInputs ++ [
```

```suggestion
  nativeBuildInputs = [
```

1. Be explicit.
2. Why??

---

OK, it is explicit now. There is a big overlap between runtime and build time deps because YottaDB is a compiler, and it compiles itself and various library code at build time, but also compiles things at runtime during its normal production operation. I can revisit the dependencies, but it is a bit messy to keep the list duplicated.